contrib/mesos/pkg/executor/executor.go

```diff
@@ -208,23 +210,74 @@ func (k *KubernetesExecutor) isDone() bool {
 	}
 }
 
+func (k *KubernetesExecutor) NodeInfo() (ni NodeInfo) {
+	k.nodeInfoLock.Lock()
+	defer k.nodeInfoLock.Unlock()
+
+	// clone k.nodeInfo
+	ni = k.nodeInfo
+	ni.Labels = map[string]string{}
+	for k, v := range k.nodeInfo.Labels {
+		ni.Labels[k] = v
+	}
+
+	return
+}
+
+func (k *KubernetesExecutor) updateNodeInfo(slaveInfo *mesos.SlaveInfo) {
+	k.nodeInfoLock.Lock()
+	defer k.nodeInfoLock.Unlock()
+
+	// TODO(sttts): when executor and kubelet are independent processes, update the Node api object here
+	// Then the ugly locking will go away.
+
+	k.nodeInfo.Labels = podtask.SlaveAttributesToLabels(slaveInfo.Attributes)
+
+	for _, r := range slaveInfo.GetResources() {
+		if r == nil || r.GetType() != mesosproto.Value_SCALAR {
+			continue
+		}
+
+		// TODO(sttts): use custom executor CPU and mem values here, not the defaults
+		switch r.GetName() {
+		case "cpus":
+			k.nodeInfo.Cores = int(r.GetScalar().GetValue())
+		// We intentionally ignore DefaultExecutorCPU here because cores are integers
+		// and we would loose a complete cpu here. This is not dramatic though
+		// because ExecutorCPU is only set to the minumm allowed of 0.01 and
+		// it's only about shares anyway, no hard cpu limit.
+		case "mem":
+			k.nodeInfo.Mem = int64(r.GetScalar().GetValue()-float64(sservice.DefaultExecutorMem)) * 1024 * 1024
```

Comment:
instead of defaultExecutorMem, can't we look at the "mem" resource of the ExecutorInfo?